tests/rptest/services/flink.py

```diff
@@ -249,17 +249,24 @@ def __init__(self, context, redpanda, topic, *args, **kwargs):
             self.STATE_FINISHED, self.STATE_CANCELED
         ]
         # Getting instance metadata
-        self.logger.info("Queriyng metadata for flink instance "
-                         f"'{self.nodes[0].account.hostname}'")
-        _provider = context.globals['cloud_provider'].upper()
-        _meta = get_metadata_for_node(_provider, self.nodes[0])
-
-        # Hardcoded xlarge instance values
         specs = instance_specs["default"]
-        if _meta['instance-type'] in instance_specs:
-            # No such type, use default
-            self.logger.info(f"Updating specs for '{_meta['instance-type']}'")
-            specs = instance_specs[_meta['instance-type']]
+        if 'cloud_provider' not in context.globals.keys():
+            self.logger.info("Can't detect cloud provider, using 'default'")
+            self.logger.debug("'cloud_provider not found in globals.json. "
+                              "Keys available: "
+                              f"{(', '.join(context.globals.keys()))}")
+        else:
+            _provider = context.globals['cloud_provider'].upper()
+            self.logger.info("Queriyng metadata for flink instance "
+                             f"'{self.nodes[0].account.hostname}'")
+            _meta = get_metadata_for_node(_provider, self.nodes[0])
+
+            # Hardcoded xlarge instance values
+            if _meta['instance-type'] in instance_specs:
+                # No such type, use default
+                self.logger.info(
+                    f"Updating specs for '{_meta['instance-type']}'")
+                specs = instance_specs[_meta['instance-type']]
```

Comment:
nit: I feel like this method doesn't belong in flink service.. move it elsewhere in some utils?

I think flink should query for total memory / vcpus and the code shouldn't really care about instance types (and error handling) here.